website/docs/tutorial/03-adding-payments.md

```diff
@@ -0,0 +1,216 @@
++use stellar_registry::import_asset;
++import_asset!(xlm);
+```
+
+Stellar Registry integrates with Scaffold Stellar, giving names & versions to contracts & contract Wasms. It also provides helpers like `import_asset` to make it easier to work with [Stellar Asset Contracts](https://developers.stellar.org/docs/tokens/stellar-asset-contract).
+
+## Step 2: 💰 Add Funds to the Contract
+
+Whenever the admin resets the number, we need to transfer some funds to the contract to get the pot started. The easiest way to do this is directly in the `set_random_number` method. Remember, this is the private function we call once in the constructor when we first deploy the contract and again any time the reset method is invoked.
+
+```rust
+    fn set_random_number(env: &Env) {
+        let new_number: u64 = env.prng().gen_range(1..=10);
+        env.storage().instance().set(&THE_NUMBER, &new_number);
+
+        // Seed the initial pot
+        let x = xlm::client(env);
+        let admin = Self::admin(env).expect("admin not set");
+        x.transfer(10_000_000_0, &admin, env.current_contract_address());
+    }
+```
+
+This creates a client to interact with the XLM contract via cross-contract calls. It gets the admin's address from storage, and then runs a transfer. If the transfer fails, perhaps because the admin does not have sufficient balance, the whole transaction gets rolled back. If this is the call to `__constructor` during the initial deploy, then the deploy will fail.
+
+You may have noticed that the number there looks really big! Seven zeroes after that `10`. When transferring assets in smart contracts, you must use their smallest-divisible unit. For XLM, this means adding seven zeroes. (The smallest unit of XLM is called a [stroop](https://developers.stellar.org/docs/learn/glossary#stroop).)
+
+## Step 3: 🙋 Update the Guess Function
+
+This is the big one! Let's make guessing cost money and pay out winners:
+
+```rust
+/// Guess a number between 1 and 10
+/// Costs a fee and pays out the entire pot if correct
+pub fn guess(env: &Env, guesser: Address, a_number: u64) -> bool {
+  let xlm_client = xlm::token_client(env);
+  let contract_address = env.current_contract_address();
+  let guessed_it = a_number == Self::number(env);
+
+  if guessed_it {
+      let balance = xlm_client.balance(&contract_address);
+      if balance == 0 {
+          panic!("Pot already won! New game not yet started.")
+      }
+
+      // pay full pot to `guesser`, whether they sent the transaction or not
+      let tx = xlm_client.transfer(
+        self.current_contract_address(),
+        guesser,
+        x.balance(self.current_contract_address()),
+      );
+      if tx.is_err {
+        panic!("transfer failed!");
+      }
+  } else {
+    // Before transferring their funds, make sure guesser is actually the one calling this function
+    guesser.require_auth();
+    let tx = xlm_client.transfer(guesser, self.current_contract_address(), 1_000_000_0);
```

Comment:
We can actually make guess without paying anything because we check the guess before taking money out. They just need to use an empty account and play until they win the pot.

We can either leave it like that and ask people if they can notice a big security issue or fix it with some explanations here.

(For an easy approach, I think we need to do the transfer out, check, and then get the pot if we are correct.)